### PR TITLE
Fix Fish shell installation

### PR DESCRIPTION
**Pull request by andreiborisov**
Fixes https://github.com/microsoft/vscode-dev-containers/issues/1501

### CODE REVIEW

**Comment by eitsupi** [comment]
Does this change fix #1501?
[The log](https://github.com/microsoft/vscode-dev-containers/runs/6867998358?check_suite_focus=true#step:4:19235) show that the fish installation was not successful?

**Comment by eitsupi** [comment]
I think the fish installation was successful because I see `Installing Fisher...` in the log.
What is failing is the installation of Fisher.

https://github.com/microsoft/vscode-dev-containers/blob/172a918f40e31bd24da8e64135026ec9f26c91b0/script-library/fish-debian.sh#L85-L88

image

**Comment by andreiborisov** [comment]
@eitsupi the issues was `check_packages fish` not working properly and as a result the script installing ancient fish version from the system package list. It has a slightly different syntax, hence `fisher` fails to work properly since it only supports fish v3 and greater.

tldr: it should fix it.

**Comment by eitsupi** [comment]
That makes sense. Thank you for working on it!

This change should be revert in this PR, I think.
https://github.com/microsoft/vscode-dev-containers/pull/1500/files#diff-b4fcdb4d2b5b4dd46c896cb5bfa0a42e633340c263aa0238d71dce261bbd5c36R112

And version check e.x. `fish -v` in this script maybe useful for debug?

**Comment by andreiborisov** [comment]
@eitsupi good suggestion! Added version check.

**Comment by andreiborisov** [comment]
@joshspicer seem to be good for the merge🙌🏻

**Comment by joshspicer** [comment]
Thank you guys for your continued support! 🚀
